core/client/src/main/java/alluxio/client/block/BlockWorkerClient.java

```diff
@@ -378,11 +379,7 @@ public Boolean call() throws AlluxioTException, TException {
         }
       });
     } catch (AlluxioException e) {
-      if (e instanceof WorkerOutOfSpaceException) {
-        return false;
-      } else {
-        throw new IOException(e);
-      }
+      throw new IOException(e);
```

Comment:
I feel this is not a back-compatible change, since it changed the protocol, and the user who previously checking on `false` for space exhaustion now needs to catch the exception.